predict.py

```diff
@@ -0,0 +1,157 @@
+               help="choose model type")
+    @cog.input("image", type=Path, help="input facial image")
+    def predict(self, model, image):
+        opts = self.opts[model]
+        opts = Namespace(**opts)
+        pprint.pprint(opts)
+
+
+        net = pSp(opts)
+        net.eval()
+        net.cuda()
+        print('Model successfully loaded!')
+
+        original_image = Image.open(str(image))
+        if opts.label_nc == 0:
+            original_image = original_image.convert("RGB")
+        else:
+            original_image = original_image.convert("L")
+        original_image.resize((256, 256))
+
+        # Align Image
+        if model not in ["celebs_sketch_to_face", "celebs_seg_to_face"]:
+            input_image = run_alignment(str(image))
+        else:
+            input_image = original_image
+
+        img_transforms = self.transforms[model]
+        transformed_image = img_transforms(input_image)
+
+        if model in ["celebs_sketch_to_face", "celebs_seg_to_face"]:
+            latent_mask = [8, 9, 10, 11, 12, 13, 14, 15, 16, 17]
+        else:
+            latent_mask = None
+
+        with torch.no_grad():
+            result_image = run_on_batch(transformed_image.unsqueeze(0), net, latent_mask)[0]
+        input_vis_image = log_input_image(transformed_image, opts)
+        output_image = tensor2im(result_image)
+
+        if model == "celebs_super_resolution":
+            res = np.concatenate([np.array(input_vis_image.resize((256, 256))),
+                                  np.array(output_image.resize((256, 256)))], axis=1)
+        else:
+            res = np.array(output_image.resize((256, 256)))
+
+        out_path = Path(tempfile.mkdtemp()) / "out.png"
+        Image.fromarray(np.array(res)).save(str(out_path))
+        return out_path
+
+
+def run_alignment(image_path):
+    import dlib
+    from scripts.align_all_parallel import align_face
+    predictor = dlib.shape_predictor("shape_predictor_68_face_landmarks.dat")
+    aligned_image = align_face(filepath=image_path, predictor=predictor)
+    print("Aligned image has shape: {}".format(aligned_image.size))
+    return aligned_image
+
+
+def run_on_batch(inputs, net, latent_mask=None):
+    if latent_mask is None:
+        result_batch = net(inputs.to("cuda").float(), randomize_noise=False)
+    else:
+        result_batch = []
+        for image_idx, input_image in enumerate(inputs):
+            # get latent vector to inject into our input image
+            vec_to_inject = np.random.randn(1, 512).astype('float32')
+            _, latent_to_inject = net(torch.from_numpy(vec_to_inject).to("cuda"),
+                                      input_code=True,
+                                      return_latents=True)
+            # get output image with injected style vector
+            res = net(input_image.unsqueeze(0).to("cuda").float(),
+                      latent_mask=latent_mask,
+                      inject_latent=latent_to_inject)
```

Comment:
I would add a flag `resize=False` here to keep the images at full resolution.  
Then in line `124` we won't need to do the resize to `256`.